lib/trino-parquet/src/main/java/io/trino/parquet/ParquetReaderOptions.java

```diff
@@ -118,129 +133,98 @@ public DataSize getSmallFileThreshold()
-                useColumnIndex,
-                useBloomFilter,
-                smallFileThreshold,
-                vectorizedDecodingEnabled);
-    }
-
-    public ParquetReaderOptions withMaxBufferSize(DataSize maxBufferSize)
-    {
-        return new ParquetReaderOptions(
-                ignoreStatistics,
-                maxReadBlockSize,
-                maxReadBlockRowCount,
-                maxMergeDistance,
-                maxBufferSize,
-                useColumnIndex,
-                useBloomFilter,
-                smallFileThreshold,
-                vectorizedDecodingEnabled);
-    }
-
-    public ParquetReaderOptions withUseColumnIndex(boolean useColumnIndex)
-    {
-        return new ParquetReaderOptions(
-                ignoreStatistics,
-                maxReadBlockSize,
-                maxReadBlockRowCount,
-                maxMergeDistance,
-                maxBufferSize,
-                useColumnIndex,
-                useBloomFilter,
-                smallFileThreshold,
-                vectorizedDecodingEnabled);
-    }
-
-    public ParquetReaderOptions withBloomFilter(boolean useBloomFilter)
-    {
-        return new ParquetReaderOptions(
-                ignoreStatistics,
-                maxReadBlockSize,
-                maxReadBlockRowCount,
-                maxMergeDistance,
-                maxBufferSize,
-                useColumnIndex,
-                useBloomFilter,
-                smallFileThreshold,
-                vectorizedDecodingEnabled);
-    }
-
-    public ParquetReaderOptions withSmallFileThreshold(DataSize smallFileThreshold)
-    {
-        return new ParquetReaderOptions(
-                ignoreStatistics,
-                maxReadBlockSize,
-                maxReadBlockRowCount,
-                maxMergeDistance,
-                maxBufferSize,
-                useColumnIndex,
-                useBloomFilter,
-                smallFileThreshold,
-                vectorizedDecodingEnabled);
-    }
-
-    public ParquetReaderOptions withVectorizedDecodingEnabled(boolean vectorizedDecodingEnabled)
-    {
-        return new ParquetReaderOptions(
-                ignoreStatistics,
-                maxReadBlockSize,
-                maxReadBlockRowCount,
-                maxMergeDistance,
-                maxBufferSize,
-                useColumnIndex,
-                useBloomFilter,
-                smallFileThreshold,
-                vectorizedDecodingEnabled);
+    public static class Builder
+    {
+        private boolean ignoreStatistics;
+        private DataSize maxReadBlockSize;
+        private int maxReadBlockRowCount;
+        private DataSize maxMergeDistance;
+        private DataSize maxBufferSize;
+        private boolean useColumnIndex;
+        private boolean useBloomFilter;
+        private DataSize smallFileThreshold;
+        private boolean vectorizedDecodingEnabled;
+
+        private Builder(ParquetReaderOptions parquetReaderOptions)
```

Comment:
this constructor is smelly. Why/where is it needed?
Maybe it's possible to move logic from `private ParquetReaderOptions()` here.